public_html/wp-content/plugins/wordcamp-reports/views/public/payment-activity.php

```diff
@@ -47,7 +47,14 @@
 
 		<div class="field_wordcamp-id">
 			<label for="wordcamp-id">WordCamp <span>(optional)</span></label>
-			<?php echo get_wordcamp_dropdown( 'wordcamp-id', array(), $wordcamp_id ); ?>
+			<?php if ( ! isset( $_GET[ 'action' ] ) ) {
+				// Show only latest 100 WordCamps by default
+				echo get_wordcamp_dropdown( 'wordcamp-id', array( 'posts_per_page' => 100, 'order' => 'DESC' ), $wordcamp_id );
```

Comment:
🤔 Are there other reports that we should make this change to? Why would `payment-activity` be the only one needed when they all call `get_wordcamp_dropdown()` with the default `$query_options`? Ah, searching for `in:#dotorg-wordcamp-logs peak memory usage` does show others having issues.

Maybe the change should be deeper, in `get_wordcamp_dropdown()` itself? Or more DRY, like by having all the `wordcamp-reports/` call it with a `WordCamp\Reports\Report\Base::get_dropdown_options()` argument?

It might also be worth considering reverting e058b4bb109c134fb006d8f1c6380a05171910e1 , so that the logs aren't cluttered with false positives. I enabled that after `95`% was cleaned up, just to see if there were others getting close, but `90` may be too low in practice.

Reply:
> 🤔 Are there other reports that we should make this change to?

Yes, I was mainly limiting the impact of any change being suggested here - should have said that in the original issue